src/transactionsFactories/transferTransactionsFactory.ts

```diff
@@ -104,17 +100,17 @@ export class TransferTransactionsFactory {
         sender: IAddress;
         receiver: IAddress;
         nativeAmount: bigint;
-        data?: string;
+        data?: Uint8Array;
     }): Transaction {
         this.ensureMembersAreDefined();
 
-        const data = options.data || "";
+        const data = options.data || new Uint8Array();
 
         return new TransactionBuilder({
             config: this.config!,
             sender: options.sender,
             receiver: options.receiver,
-            dataParts: [data],
+            dataParts: [Buffer.from(data).toString()],
```

Comment:
Here might be an issue. Since `dataParts` wants strings - but `Buffer.from(data).toString()` is not necessarily a valid / utf8 string. Can we create the transaction without `TransactionBuilder`? Or perhaps just set the data after creating it with the builder?

Reply:
I think `Buffer.toString()` returns by default a `utf8` string.